src/tally.F90

```diff
@@ -27,524 +27,573 @@ module tally
-! energy range, etc. Note that since these are "analog" tallies, they are only
-! triggered at every collision, not every event
+! SCORE_GENERAL adds scores to the tally array for the given filter and nuclide.
+! This will work for either analog or tracklength tallies.  Note that
+! atom_density and flux are not used for analog tallies.
 !===============================================================================
 
-  subroutine score_analog_tally(p)
+  subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
+       &atom_density, flux)
+    type(Particle),             intent(in)    :: p
+    type(TallyObject), pointer, intent(inout) :: t
+    integer,                    intent(in)    :: start_index
+    integer,                    intent(in)    :: i_nuclide
+    integer,                    intent(in)    :: filter_index
+    real(8),                    intent(in)    :: flux
+    real(8),                    intent(in)    :: atom_density   ! atom/b-cm
 
-    type(Particle), intent(in) :: p
-
-    integer :: i
-    integer :: i_tally
     integer :: j                    ! loop index for scoring bins
-    integer :: k                    ! loop index for nuclide bins
+    integer :: l                    ! loop index for nuclides in material
+    integer :: m                    ! loop index for reactions
     integer :: n                    ! loop index for legendre order
     integer :: num_nm               ! Number of N,M orders in harmonic
-    integer :: l                    ! scoring bin loop index, allowing for changing
-                                    ! position during the loop
-    integer :: filter_index         ! single index for single bin
+    integer :: q                    ! loop index for scoring bins
+    integer :: i_nuc                ! index in nuclides array (from material)
+    integer :: i_energy             ! index in nuclide energy grid
     integer :: score_bin            ! scoring bin, e.g. SCORE_FLUX
-    integer :: i_nuclide            ! index in nuclides array
     integer :: score_index          ! scoring bin index
+    real(8) :: atom_density_        ! atom/b-cm
+    real(8) :: f                    ! interpolation factor
     real(8) :: score                ! analog tally score
-    real(8) :: last_wgt             ! pre-collision particle weight
-    real(8) :: wgt                  ! post-collision particle weight
-    real(8) :: mu                   ! cosine of angle of collision
+    real(8) :: score_               ! analog tally score
     real(8) :: macro_total          ! material macro total xs
     real(8) :: macro_scatt          ! material macro scatt xs
-    logical :: found_bin            ! scoring bin found?
-    type(TallyObject), pointer, save :: t => null()
-!$omp threadprivate(t)
+    real(8) :: uvw(3)               ! particle direction
+    type(Material),    pointer, save :: mat => null()
+    type(Reaction),    pointer, save :: rxn => null()
+!$omp threadprivate(t, mat, rxn)
```

Comment:
I'm not familiar with OpenMP so I'd appreciate some help with this.  It looks like `t` shouldn't be declared threadprivate, but I get a test_confidence_intervals failure if I remove it from the threadprivate declaration.